src/main/java/io/sealights/plugins/sealightsjenkins/buildsteps/cli/SealightsCLIBuildStep.java

```diff
@@ -33,6 +38,44 @@ public SealightsCLIBuildStep(boolean enabled, boolean failBuildIfStepFail,
         this.cliRunner = cliRunner;
     }
 
+    /* * The goal of this method is to support migration of data between versions
+    * of this plugin.
+    */
+    private Object readResolve() {
+        if (cliRunner != null) {
+            return resolveFromCLiRunner();
+        }
+        return this;
+    }
+
+    private Object resolveFromCLiRunner() {
+        StringBuffer additionalArgs = new StringBuffer();
+        if (this.commandMode instanceof CommandMode.ConfigView) {
+            if (!StringUtils.isNullOrEmpty(cliRunner.getAppName()))
+                ((CommandMode.ConfigView) commandMode).setAppName(cliRunner.getAppName());
+            if (!StringUtils.isNullOrEmpty(cliRunner.getBranchName()))
+                ((CommandMode.ConfigView) commandMode).setBranchName(cliRunner.getBranchName());
+            if (cliRunner.getBuildName() != null)
+                ((CommandMode.ConfigView) commandMode).setBuildName(cliRunner.getBuildName());
+        } else if (this.commandMode instanceof CommandMode.EndView ||
+                this.commandMode instanceof CommandMode.StartView ||
+                this.commandMode instanceof CommandMode.ExternalReportView ||
+                this.commandMode instanceof CommandMode.UploadReportsView) {
+            if (!StringUtils.isNullOrEmpty(cliRunner.getAppName())) {
+                additionalArgs.append("appName=" + cliRunner.getAppName() + "\n");
+            }
+            if (!StringUtils.isNullOrEmpty(cliRunner.getBranchName())) {
+                additionalArgs.append("branchName=" + cliRunner.getBranchName() + "\n");
+            }
+
```

Comment:
what about buildname? use 
`buildNameResolver.getFinalBuildName(build, envVars, buildName, logger))` to resolve the build name string